src/lib/bootloader/proposal_client.rb

```diff
@@ -197,6 +195,32 @@ def construct_proposal_map
       ret
     end
 
+    # Result of {#make_proposal} if a {Bootloader::NoRoot} exception is raised
+    # while calculating the proposal
+    #
+    # @return [Hash]
+    def no_root_proposal
+      {
+        "label_proposal" => [],
+        "warning_level"  => :fatal,
+        "warning"        => _("Cannot detect device mounted as root. Please check partitioning.")
+      }
+    end
+
+    # Result of {#make_proposal} if a {Bootloader::BrokenConfiguration} exception
+    # is raised while calculating the proposal
+    #
+    # @param err [Bootloader::BrokenConfiguration] raised exception
+    # @return [Hash]
+    def broken_configuration_proposal(err)
+      {
+        "label_proposal" => [],
+        "warning_level"  => :error,
+        # TRANSLATORS: %s is a string containing the technical details of the error
+        "warning"        => _("Error reading the bootloader configuration files. %s.") % err.reason
```

Comment:
well, this is kind of new as it can happen only during upgrade where we read previous configuration.
But it should contain what user can do? Ideally user should click on "Booting" and here he should be able to repropose configuration from scratch, same as during runtime.

So I suggest to adapt error message here and also allow user to repropose after clicking on Booting.

Reply:
What I'm fixing right now is that if you click on "Booting" the error pops-up. So the exception has to be captured in a third place.